flake.nix

```diff
@@ -3,35 +3,85 @@
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
-    flake-compat = { url = "github:edolstra/flake-compat"; flake = false; };
-    libnbtplusplus = { url = "github:PrismLauncher/libnbtplusplus"; flake = false; };
+    flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks = {
+      url = "github:cachix/pre-commit-hooks.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.flake-utils.follows = "flake-utils";
+    };
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    libnbtplusplus = {
+      url = "github:PrismLauncher/libnbtplusplus";
+      flake = false;
+    };
   };
 
-  outputs = { self, nixpkgs, libnbtplusplus, ... }:
-    let
-      # User-friendly version number.
-      version = builtins.substring 0 8 self.lastModifiedDate;
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    pre-commit-hooks,
+    libnbtplusplus,
+    ...
+  }: let
+    # User-friendly version number.
+    version = builtins.substring 0 8 self.lastModifiedDate;
+
+    # Supported systems (qtbase is currently broken for "aarch64-darwin")
+    supportedSystems = with flake-utils.lib.system; [
+      x86_64-linux
+      x86_64-darwin
+      aarch64-linux
+    ];
+
+    packagesFn = pkgs: rec {
```

Comment:
i don't think `rec` is needed here

Reply:
I am not sure. I have seen this a lot, not just here but also in nixpkgs, so I don't know if it's convention or anything. I removed it anyway.